### ISSUE TITLE
Add freebsd CI

### ISSUE THREAD

**Issue by trufae** [radareorg/radare2#18597]
https://github.com/vmactions/freebsd-vm

**Comment by koobs** [comment]
See also Cirrus CI, which generously provides CI free for OpenSource projects. Don't hesitate to reach out if you need tips/pointers

**Comment by trufae** [comment]
Im quite interesting into having r2 in CI building for freebsd, netbsd and openbsd. I am setting up a build farm at home, and i tried to simplify the current CI to just use GHA for maintainance reasons. if someone is willing to submit a PR adding the cirrus CI for freebsd it will be good to go. i can add the settings needed in the r2 side.

Do you know if its possible run net and open in cirrus? maybe using an emulation layer on top of freebsd?

thanks

**Comment by koobs** [comment]
Cirrus supports OS's that provide images in GCE, and I see open issues for at least requesting OpenBSD support (so, not yet it seems).

I'll try to organise a PR adding Cirrus/FreeBSD for you, and I also provide FreeBSD buildbot workers for several upstream incl Python, and would be happy to provide one for you too. Just let me know if that would be valuable.

**Comment by koobs** [comment]
OpenBSD and NetBSD on FreeBSD via bhuve may be possible, but likely not the least amount of long-term maintenance. I'd reco GHA+Cirrus+lightweight CI you run (or a SaaS), where people can provide remote workers for other OS's

**Comment by trufae** [comment]
That sounds perfect! looking forward your PR. btw just curious about which versions of freebsd are supported in cirrus :? i think some people would be interested in testing r2 on some old versions too.

**Comment by koobs** [comment]
> 
> 
> That sounds perfect! looking forward your PR. btw just curious about which versions of freebsd are supported in cirrus :? i think some people would be interested in testing r2 on some old versions too.

All supported branches (currently 11.x, 12,x 13,x) + development branch (14-CURRENT), but the basis is whichever our (FreeBSD's) release engineering push to GCE. I can make any specific requests for other versions if and when necessary, including for !x86 archs if supported in GCE

**Comment by meme** [comment]
Have you considered sr.ht: https://man.sr.ht/builds.sr.ht/compatibility.md? It supports FreeBSD and OpenBSD

**Comment by devnexen** [comment]
seems to have been removed since ? not criticized it but as I try to find CI for openbsd wanted to know the legitimate concern about sr ht.

**Comment by trufae** [comment]
Yes. it was removed in 3a5c35f68fc610c6862676488d4e0d347ec3006a because it was no longer working because the account wasn't created by me so i couldnt access. could be added back and im totally +1 for it
